From afadd8fc69b1057dcba5082fa3657d080bdde11c Mon Sep 17 00:00:00 2001
From: Rama Chavali <rama.rao@salesforce.com>
Date: Fri, 12 Nov 2021 14:26:23 +0530
Subject: do not generate debug output for ndsz if dns is not enabled (#36043)

* do not generate debug output for ndsz if dns is not enabled

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* handle nil connection

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* move down

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>
---
 pilot/pkg/xds/debug.go | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/pilot/pkg/xds/debug.go b/pilot/pkg/xds/debug.go
index a9e01d0a4d..0eaeae9c41 100644
--- a/pilot/pkg/xds/debug.go
+++ b/pilot/pkg/xds/debug.go
@@ -784,12 +784,16 @@ func (s *DiscoveryServer) ndsz(w http.ResponseWriter, req *http.Request) {
 	if s.handlePushRequest(w, req) {
 		return
 	}
-
 	proxyID, con := s.getDebugConnection(req)
 	if con == nil {
 		s.errorHandler(w, proxyID, con)
 		return
 	}
+	if !con.proxy.Metadata.DNSCapture {
+		w.WriteHeader(http.StatusBadRequest)
+		_, _ = w.Write([]byte("DNS capture is not enabled in the proxy\n"))
+		return
+	}
 
 	if s.Generators[v3.NameTableType] != nil {
 		nds, _, _ := s.Generators[v3.NameTableType].Generate(con.proxy, s.globalPushContext(), nil, nil)
-- 
2.35.3

